### PR TITLE
Add App Grid Dashboard to TMail web

### PR DESCRIPTION
**Pull request by hoangdat**
- [x] configuration loader
- [x] show for desktop web

https://user-images.githubusercontent.com/6462404/195907069-4517eac5-5044-4102-8194-7744f2f424c3.mov

- [ ] show for tablet web
- [ ] show for mobile web
- [ ] documentation to deploy with  configuration

### CODE REVIEW

**Comment by chibenwa** [comment]
Awesome!

Is there an example of how the configuration looks like?

Also I think we can remove INBOX from the app grid, add Jitsi,

**Comment by chibenwa** [comment]
By the way I just want to make sure that one do not need to recompile the application in order to change the app grid...

**Comment by chibenwa** [comment]
Also if the app grid is empty (no other application) do we still display it?

**Comment by hoangdat** [comment]
> By the way I just want to make sure that one do not need to recompile the application in order to change the app grid...

will testing with docker images when all the tasks done

File: configurations/app_dashboard.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
+  "apps": [
```
Comment (dab246) [inline configurations/app_dashboard.json:2]: good idea

File: configurations/app_dashboard.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    {
+      "appName": "Twake",
```
Comment (dab246) [inline configurations/app_dashboard.json:4]: Do we need to care about multiple languages? @hoangdat

File: lib/features/mailbox_dashboard/domain/app_dashboard/app_dashboard_configuration_parser.dart [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:typed_data';
+import 'package:core/utils/app_logger.dart';
+import 'package:core/utils/config/app_config_parser.dart';
+import 'package:tmail_ui_user/features/mailbox_dashboard/domain/app_dashboard/linagora_applications.dart';
+
+class AppDashboardConfigurationParser extends AppConfigParser<LinagoraApplications> {
+  @override
+  Future<LinagoraApplications> parse(String value) async {
```
Comment (dab246) [inline lib/features/mailbox_dashboard/domain/app_dashboard/app_dashboard_configuration_parser.dart:10]: remove `async`  if you are not use `await` in function

File: lib/features/mailbox_dashboard/presentation/controller/app_grid_dashboard_controller.dart [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import 'package:core/presentation/state/failure.dart';
+import 'package:core/presentation/state/success.dart';
+import 'package:core/utils/app_logger.dart';
+import 'package:dartz/dartz.dart';
+import 'package:get/get_rx/get_rx.dart';
+import 'package:tmail_ui_user/features/mailbox_dashboard/domain/app_dashboard/linagora_applications.dart';
+import 'package:tmail_ui_user/features/mailbox_dashboard/domain/usecases/get_app_dashboard_configuration_interactor.dart';
+import 'package:tmail_ui_user/main/utils/app_config.dart';
+
+class AppGridDashboardController {
```
Comment (dab246) [inline lib/features/mailbox_dashboard/presentation/controller/app_grid_dashboard_controller.dart:10]: why you are not extend from `GetxController`

File: configurations/app_dashboard.json (hunk 2 above)
Comment (hoangdat) [reply configurations/app_dashboard.json:4]: no, app name is unique in all languages

File: lib/features/mailbox_dashboard/presentation/controller/app_grid_dashboard_controller.dart (hunk 4 above)
Comment (hoangdat) [reply lib/features/mailbox_dashboard/presentation/controller/app_grid_dashboard_controller.dart:10]: no need at the moment, so no implement :)

**Review by dab246 — APPROVED** [review]

**Review by ManhNTX — APPROVED** [review]

File: lib/features/mailbox_dashboard/domain/app_dashboard/linagora_app.dart [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import 'package:equatable/equatable.dart';
+import 'package:json_annotation/json_annotation.dart';
+
+part 'linagora_app.g.dart';
+
+@JsonSerializable()
+class LinagoraApp with EquatableMixin{
+  @JsonKey(name: 'appName')
```
Comment (sherlockvn) [inline lib/features/mailbox_dashboard/domain/app_dashboard/linagora_app.dart:8]: I think this is not necessary.

File: lib/features/mailbox_dashboard/domain/app_dashboard/linagora_applications.dart [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import 'package:equatable/equatable.dart';
+import 'package:json_annotation/json_annotation.dart';
+import 'linagora_app.dart';
+
+part 'linagora_applications.g.dart';
+
+@JsonSerializable()
+class LinagoraApplications with EquatableMixin {
+  @JsonKey(name: 'apps')
```
Comment (sherlockvn) [inline lib/features/mailbox_dashboard/domain/app_dashboard/linagora_applications.dart:9]: same.

**Pull request merged**